src/power_grid_model_io/converters/tabular_converter.py

```diff
@@ -179,8 +216,22 @@ def _convert_col_def_to_attribute(
         component: str,
         attr: str,
         col_def: Any,
-        extra_info: Optional[ExtraInfoLookup],
+        extra_info: Optional[ExtraInfoLookup] = None,
```

Comment:
I understand this urge, but I try not to use default values for private functions. I think it's better to be more explicit and let the caller explicitly state `extra_info=None`. Currently this function is only called from `_convert_table_to_component()`, which forwards the `extra_info` attribute it receives, that function receives it from `_parse_data()`* and that function receives it from `load_input_data()`, `load_update_data()` etc. By providing a default value, one might forget to forward the `extra_info` attribute.

In those high level functions, it is decided that `extra_info` is only loaded for input data. I'm not sure that that's correct, but because I found it too complex to handle extra_info for batch datasets, I decided to implement it only for input data for now, and we can think of more complex cases later on.

* Note that `_parse_data()` also has a default `None` value for the `extra_info` attribute, but it shouldn't.

Reply:
@petersalemink95, how do you think about this? In general I would like to avoid default values for attributes of private methods.